src/main/java/io/github/bobocodebreskul/context/registry/BringContainer.java

```diff
@@ -0,0 +1,86 @@
+package io.github.bobocodebreskul.context.registry;
+
+import io.github.bobocodebreskul.context.config.BeanDefinition;
+import io.github.bobocodebreskul.context.exception.InstanceCreationException;
+import io.github.bobocodebreskul.context.exception.NoSuchBeanDefinitionException;
+import io.github.bobocodebreskul.context.exception.NoSuchMethodRuntimeException;
+import io.github.bobocodebreskul.context.scan.RecursiveClassPathAnnotatedBeanScanner;
+import io.github.bobocodebreskul.context.scan.utils.ScanUtilsImpl;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Implementation of the {@link ObjectFactory} as Bring beans container. Creates and holds
+ * all found and registered beans.
+ *
+ * @author Ruslan Hladchenko
+ * @author Roman Pryshchepa
+ * @author Mykola Filimonov
+ * @author Vitalii Katkov
+ */
+@Slf4j
+public class BringContainer implements ObjectFactory {
+
+  private final Map<Class<?>, Object> storageByClass = new ConcurrentHashMap<>(16);
```

Comment:
I'd recommend to store this magic number '16' as a separate default value for storage initial size, for example in utility class GeneralConstants